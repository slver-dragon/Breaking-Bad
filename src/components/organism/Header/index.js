import React from "react";
import { Link, useLocation } from "react-router-dom";
import style from "./style.module.scss";
import logo from "../../../assets/img/logo.svg";

const Header = () => {
  const catalogStyle = (useLocation().pathname === "/catalog") 
       ? ( style.catalog + ' ' + style.forCardList )
       : ( style.catalog );
  return (
    <div className={style.main}>
      <img className={style.image} src={logo} alt="Breaking Bad" />
      <Link to={"/catalog"} className={ catalogStyle }>
        <span className={style.threeLine}>
          <i></i>
          <i></i>
          <i></i>
        </span>
        <p>Каталог</p>
      </Link>
    </div>
  );
};
export default Header;