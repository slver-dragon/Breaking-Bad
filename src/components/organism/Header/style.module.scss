@import "../../../assets/styles/style";

.main {
  display: flex;
  justify-content: space-between;
  @include mediumFont;
  @include mainContainer;
  @include headerMain;
  align-items: center;
  .catalog {
    display: block;
    border: 0.0875rem solid $yellow;
    border-radius: 0.25rem;
    cursor: pointer;
    margin: 0.25rem 0;
    padding: 0.45rem 1rem 0.3rem;
    text-decoration: none;
    color: $black;
  }
  .forCardList {
    background-color: $yellow;
  }
}
