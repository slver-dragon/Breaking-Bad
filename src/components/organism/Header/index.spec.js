import React from "react";
import { mount } from "enzyme";
import Header from ".";
import { BrowserRouter } from "react-router-dom/cjs/react-router-dom.min";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "/catalog"
  }),
}));

describe("Testing organism Header:", () => {
  const component = mount(<BrowserRouter><Header /></BrowserRouter>);

  it("is rendering", () => {    
    expect(component).toMatchSnapshot();
  });
});
