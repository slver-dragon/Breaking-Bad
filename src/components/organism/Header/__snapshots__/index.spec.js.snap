// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing organism Header: is rendering 1`] = `
<e>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Header>
      <div
        className="main"
      >
        <Link
          className="image"
          to="/"
        >
          <LinkAnchor
            className="image"
            href="/"
            navigate={[Function]}
          >
            <a
              className="image"
              href="/"
              onClick={[Function]}
            >
              <Button
                img="test-file-stub"
                text=""
              >
                <button
                  className="button"
                >
                  <img
                    alt=""
                    src="test-file-stub"
                  />
                </button>
              </Button>
            </a>
          </LinkAnchor>
        </Link>
        <Link
          className="catalog forCardList"
          to="/catalog"
        >
          <LinkAnchor
            className="catalog forCardList"
            href="/catalog"
            navigate={[Function]}
          >
            <a
              className="catalog forCardList"
              href="/catalog"
              onClick={[Function]}
            >
              <Button
                img="test-file-stub"
                text="Каталог"
              >
                <button
                  className="button"
                >
                  <img
                    alt=""
                    src="test-file-stub"
                  />
                  <span>
                    Каталог
                  </span>
                </button>
              </Button>
            </a>
          </LinkAnchor>
        </Link>
      </div>
    </Header>
  </Router>
</e>
`;
